mailserver/mailserver_db_leveldb.go

```diff
@@ -68,7 +68,16 @@ func NewLevelDB(dataDir string) (*LevelDB, error) {
 		log.Info("database is corrupted trying to recover", "path", dataDir)
 		db, err = leveldb.RecoverFile(dataDir, nil)
 	}
-	return &LevelDB{ldb: db}, err
+	instance := LevelDB{ldb: db}
+
+	// Initialize the archive counter
+	count, err := instance.envelopesCount()
+	if err != nil {
+		return nil, err
```

Comment:
I though we wanted not to fail but log it only in the case of the error because it's not a crucial operation.